app/models/goal.py

```diff
@@ -1,5 +1,28 @@
 from app import db
+from flask import abort, make_response
 
 
 class Goal(db.Model):
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal", lazy=True)
+
+    def to_dict_goals(self):
```

Comment:
Minor naming: the `goals` part of `to_dict_goals` doesn't really had much information and even creates a bit of repetition if we imagine calling this through `goal.to_dict_goals`. I think just `to_dict` is fine here, though I also see that this is a bit different than the `to_dict` of `Task` in that it puts its result in a `"goal"` key.

That way of working suggests forming this into a _response_ so a possible way might be to name the method `to_response`!